cmd/state-svc/main.go

```diff
@@ -100,6 +104,22 @@ func run(cfg *config.Instance) error {
 		return errs.Wrap(err, "Could not initialize outputer")
 	}
 
+	if svcExec, err := installation.ServiceExec(); err == nil {
+		if as, err := autostart.New(svcAutostart.App, svcExec, nil, svcAutostart.Options, cfg); err == nil {
+			configMediator.AddListener(constants.AutostartSvcConfigKey, func() {
+				if cfg.GetBool(constants.AutostartSvcConfigKey) {
+					as.Enable()
+				} else {
+					as.Disable()
+				}
+			})
+		} else {
+			multilog.Error("state-svc could not find its autostart")
+		}
```

Comment:
Please consider hiding this within the cmd/state-svc/autostart package as a function named something like "NewAutostartAutostart". I realize there's a stutter, and am open to abbreviating or simplifying in some manner. The intention of that naming convention is to emphasize that the type returned is internal/osutils/autostart.Autostart (i.e. from another package) rather than a type named Autostart that is local to the cmd/state-svc/autostart package (which would normally have a construction func named "New", so maybe "NewAutostart" is enough in your opinion.).

Reply:
Thanks for the suggestion. I took this a step further and moved the entire block into the autostart package and exposed a single function: `RegisterConfigListener()`. We don't need a persistent object here, so we side-step the potentially awkward naming. I hope this is okay with you.